// express-responsive/lib/UserAgentMatcher.js
// Generated by CoffeeScript 1.6.2
var UserAgentMatcher;

UserAgentMatcher = (function() {
  function UserAgentMatcher(agent) {
    this.agent = agent;
  }

  UserAgentMatcher.prototype.matchTV = function() {
    if (this.agent.match(/GoogleTV|SmartTV|InternetTV|NetCast|NETTV|AppleTV|boxee|Kylo|Roku|DLNADOC|CE\-HTML/i)) {
      return true;
    }
    if (this.agent.match(/Xbox|PLAYSTATION 3|Wii/i)) {
      return true;
    }
    return false;
  };

  UserAgentMatcher.prototype.matchTablet = function() {
    if (this.agent.match(/iP(a|ro)d/i)) {
      return true;
    }
    if (this.agent.match(/tablet/i) && !this.agent.match(/RX-34/i)) {
      return true;
    }
    if (this.agent.match(/FOLIO/i)) {
      return true;
    }
    if (this.agent.match(/Linux/i) && this.agent.match(/Android/i) && !this.agent.match(/Fennec|mobi|HTCMagic|HTCX06HT|NexusOne|SC-02B|fone945/i)) {
      return true;
    }
    if (this.agent.match(/Kindle/i)) {
      return true;
    }
    if (this.agent.match(/Mac OS/i) && this.agent.match(/Silk/i)) {
      return true;
    }
    if (this.agent.match(/GT-P10|SC-01C|SHW-M180S|SGH-T849|SCH-I800|SHW-M180L|SPH-P100|SGH-I987|zt180|HTC(Flyer|_Flyer)|SprintATP51|ViewPad7|pandigital(sprnova|nova)|IdeosS7|DellStreak7|AdventVega|A101IT|A70BHT|MID7015|Next2|nook/i)) {
      return true;
    }
    if (this.agent.match(/MB511/i) && this.agent.match(/RUTEM/i)) {
      return true;
    }
    return false;
  };

  UserAgentMatcher.prototype.matchDesktop = function() {
    if (this.agent.match(/Windows (NT|XP|ME|9)/i) && !this.agent.match(/Phone/i) && !this.agent.match(/Bot|Spider|ia_archiver|NewsGator/i)) {
      return true;
    }
    if (this.agent.match(/Win( ?9|NT)/i)) {
      return true;
    }
    if (this.agent.match(/Macintosh|PowerPC/i) && !this.agent.match(/Silk/i)) {
      return true;
    }
    if (this.agent.match(/Linux/i) && this.agent.match(/X11/i) && !this.agent.match(/Charlotte|GoogleTV/i)) {
      return true;
    }
    if (this.agent.match(/CrOS/i)) {
      return true;
    }
    if (this.agent.match(/Solaris|SunOS|BSD/i)) {
      return true;
    }
    return false;
  };

  UserAgentMatcher.prototype.matchBot = function() {
    if (this.agent.match(/curl|Bot|B-O-T|Crawler|Spider|Spyder|Yahoo|ia_archiver|Covario-IDS|findlinks|DataparkSearch|larbin|Mediapartners-Google|NG-Search|Snappy|Teoma|Jeeves|Charlotte|NewsGator|TinEye|Cerberian|SearchSight|Zao|Scrubby|Qseero|PycURL|Pompos|oegp|SBIder|yoogliFetchAgent|yacy|webcollage|VYU2|voyager|updated|truwoGPS|StackRambler|Sqworm|silk|semanticdiscovery|ScoutJet|Nymesis|NetResearchServer|MVAClient|mogimogi|Mnogosearch|Arachmo|Accoona|holmes|htdig|ichiro|webis|LinkWalker|lwp-trivial/i) && !this.agent.match(/phone|Playstation/i)) {
      return true;
    }
    return false;
  };

  UserAgentMatcher.prototype.matchPhone = function() {
    if (!this.matchTV() && !this.matchTablet() && !this.matchDesktop() && !this.matchBot()) {
      return true;
    }
    return false;
  };

  return UserAgentMatcher;

})();

module.exports = UserAgentMatcher;
